Sources/DeepLearning/Layer.swift

```diff
@@ -51,6 +51,71 @@ public extension Layer
     }
 }
 
+/// A sequencing of two Layers to form a chain.
+///
+/// The sequential combinator composes two layers sequentially, feeding the
+/// output of the first layer as input to the second layer.
+///
+/// It is normally constructed via the >> operator.
+public struct Sequential<LHS: Layer, RHS: Layer>: Layer
+    where LHS.Output == RHS.Input,
+          LHS.TangentVector: AdditiveArithmetic,
+          RHS.TangentVector: AdditiveArithmetic,
+          LHS.CotangentVector: AdditiveArithmetic,
+          RHS.CotangentVector: AdditiveArithmetic,
+          LHS.Input.CotangentVector: AdditiveArithmetic,
+          LHS.Output.CotangentVector: AdditiveArithmetic,
+          RHS.Output.CotangentVector: AdditiveArithmetic,
+          RHS.Output.TangentVector: AdditiveArithmetic {
+    let lhs: LHS
+    let rhs: RHS
+
+     init(_ lhs: LHS, _ rhs: RHS) {
+        self.lhs = lhs
+        self.rhs = rhs
+    }
+
+    @differentiable(wrt: (self, input))
+    public func applied(to input: LHS.Input) -> RHS.Output {
+        let intermediateValue = lhs.applied(to: input)
+        return rhs.applied(to: intermediateValue)
+    }
+}
+
+public func >> <LHS: Layer, RHS: Layer>(_ lhs: LHS, _ rhs: RHS) -> Sequential<LHS, RHS> {
+    return Sequential(lhs, rhs)
+}
+
+// TODO(TF-244): Uncomment once TF-244 is resolved.
+//public struct Parallel<LHS: Layer, RHS: Layer, LHSOutput, RHSOutput, AggregateOutput: Differentiable>: Layer
```

Comment:
It should not be necessary to add `LHSOutput` and `RHSOutput` generic type parameters because you can refer to them as `LHS.Output` and `RHS.Output`.

Have you tried that? The type of `combiner` would then be:
```swift
let combiner: @differentiable (LHS.Output, RHS.Output) -> AggregateOutput
```

Reply:
Good point. Thanks!

I've updated the code. Haven't checked yet to see if it compiles because I'm currently running into [TF-244](https://bugs.swift.org/browse/TF-244).

Reply:
This is not the right direction. Layer identity and existence as a property are two critical part of this API. The former is for clarify, while the latter is for being *even able* to differentiate w.r.t. parameters inside chained layers.

Reply:
BTW, I do acknowledge the validity of this idea in the fullness of time. Differentiating w.r.t. inner layers' parameters will fall out of the _currying and differentiable closure contexts_ semantics that @pschuh is working on. But even then, this will not be a preferable approach because
- `AnyDerivative` existential is highly inefficient than concrete layer types. This is already worth a thousand words I suppose...
- Users can't refer to sublayers, parameters, per-sublayer, or per-parameter gradients in that model.